src/cache_cf.cc

```diff
@@ -2123,12 +2123,11 @@ Configuration::Component<CachePeers*>::Reconfigure(SmoothReconfiguration &sr, Ca
     debugs(3, 5, *newPeer);
 
     Assure(peers == Config.peers);
-    const auto currentPeer = findCachePeerByName(newPeer->name);
+    const auto currentPeer = sr.oldPeers ? sr.oldPeers->take(newPeer->name) : nullptr;
     if (currentPeer) {
-        if (!currentPeer->stale)
-            throw TextException("cache_peer specified twice", Here());
         // TODO: Consider reporting unchanged configurations.
         currentPeer->update(sr, *newPeer);
+        AddConfigured(currentPeer);
```

Comment:
I guess we need peerSelectAdd() here (as in the 'else' block), because the configuration could change. The old code did not to this for some reason...